networkdb/delegate.go

```diff
@@ -63,10 +82,27 @@ func (nDB *NetworkDB) purgeSameNode(n *node) {
 }
 
 func (nDB *NetworkDB) handleNodeEvent(nEvent *NodeEvent) bool {
-	n := nDB.checkAndGetNode(nEvent)
+	// Update our local clock if the received messages has newer
+	// time.
+	nDB.networkClock.Witness(nEvent.LTime)
+
+	n := nDB.getNode(nEvent)
```

Comment:
Currently `checkAndGetNode` gets called in `handleNodeEvent`. This function checks all the lists. Since we are adding `getNode` before `checkAndGetNode` its safer to keep the same search logic. Making it too restrictive (ie: searching only in `nDB.nodes`) and bailing out early can break some other scenario thats currently working.

Also, in this particular issue its ok even if the node is in the `failedNodes`. It can happen if there is out of order events and we have already handled `NotifyLeave`